src/components/AddTransactionForm.tsx

```diff
@@ -0,0 +1,306 @@
+      },
+      {
+        message: 'Datum musí být v rozmezí posledních 5 let a ne v budoucnosti',
+      },
+    ),
+  price: z
+    .string({ required_error: 'Prosím zadejte jednotkovou cenu' })
+    .transform((val) => parseFloat(val))
+    .refine((val) => !isNaN(val) && val > 0, {
+      message: 'Cena musí být nenulová a kladná.',
+    }),
+  currency: z.string({
+    required_error: 'Prosím zadejte měnu transakce',
+  }),
+  fee: z
+    .string()
+    .optional()
+    .transform((val) => (val?.trim() ? parseFloat(val) : 0))
+    .refine((val) => !isNaN(val) && val >= 0, {
+      message: 'Poplatky musí být nezáporné číslo',
+    }),
+});
+
+type AddTransactionFormFields = z.infer<typeof schema>;
+
+export const AddTransactionForm = () => {
+  const { symbolList, isLoading } = useSymbolListContext();
+
+  const methods = useForm<AddTransactionFormFields>({
+    resolver: zodResolver(schema),
+  });
+
+  const [selectedSymbol, setSelectedSymbol] = useState('');
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    const existingTransactions = JSON.parse(
+      localStorage.getItem('transactions') || '[]',
+    );
+    console.log('Loaded transactions:', existingTransactions);
+  }, []);
+
+  const onSubmit: SubmitHandler<AddTransactionFormFields> = (data) => {
+    try {
+      const existingTransactions = JSON.parse(
+        localStorage.getItem('transactions') || '[]',
+      );
+
+      const updatedTransactions = [...existingTransactions, data];
+
+      localStorage.setItem('transactions', JSON.stringify(updatedTransactions));
+
+      console.log('Transaction saved successfully!', updatedTransactions);
+
+      setSuccessMessage('Transakce byla úspěšně odeslána.');
+      setErrorMessage(null);
+
+      methods.reset();
+      setSelectedSymbol('');
+
+      setTimeout(() => setSuccessMessage(null), 7000);
+    } catch (error) {
+      console.error('Error saving transaction:', error);
+
+      setErrorMessage('Nastala chyba při ukládání transakce. Zkuste to znovu.');
+      setSuccessMessage(null);
+
+      setTimeout(() => setErrorMessage(null), 7000);
+    }
+  };
+
+  if (isLoading) {
+    return (
+      <div className="flex h-64 items-center justify-center">
+        <LoaderIcon className="h-8 w-8 animate-spin text-gray-500" />
+      </div>
+    );
+  }
+
+  return (
+    <FormProvider {...methods}>
+      <form
+        className="mx-auto flex max-w-sm flex-col gap-4 p-4"
+        onSubmit={methods.handleSubmit(onSubmit)}
+      >
+        {successMessage && (
+          <div className="rounded-md bg-green-100 p-3 text-green-800">
+            {successMessage}
+          </div>
+        )}
+        {errorMessage && (
+          <div className="rounded-md bg-red-100 p-3 text-red-800">
+            {errorMessage}
+          </div>
+        )}
+
+        <FormField
```

Comment:
Ideally I would advise to define each `FormField` as a separate component then import them. The form will be much more readable.